microsoft/electionguard-ui
# 🗝️ Key upload

[//]: # (🚨 Please review the CONTRIBUTING.md in this repository. 💔Thank you!)

### Issue

Fixes #121 

### Description
Replaces the joint key retrieved step with a joint key upload step, whereby you can specify a key.json file that will provide the commitment_hash, crypto_base_hash, crypto_extended_base_hash, and elgamal_public_key values that are required to create an election.

### Testing
1. Log in
2. Select Setup Election
3. Provide basic info
4. Select guardian key

Expected you should be on a Joint Upload Key page that looks like this:

![image](https://user-images.githubusercontent.com/1769905/151588271-6e980d87-89a0-4ecf-a6ee-4dbe96554492.png)

5. Click upload, select an invalid file

Expected: an error message like this:

![image](https://user-images.githubusercontent.com/1769905/151588370-405bed37-0680-4230-bd06-314d8bbd95ce.png)

6. Select a valid file (from `packages\admin-app\public\key.json`)

Expected: you should move to the next step
